script.js

```diff
@@ -0,0 +1,521 @@
+        keys[key].querySelectorAll('.lower').forEach((item) => item.classList.remove('active'));
+      }
+    }
+  }
+}
+
+function doArrowLeft() {
+  if (textarea.selectionStart !== 0) {
+    textarea.selectionStart -= 1;
+    textarea.selectionEnd -= 1;
+  }
+}
+
+function doArrowRight() {
+  textarea.selectionStart += 1;
+}
+
+let isCaps = false;
+
+function doCaps() {
+  isCaps = true;
+  document.querySelector('.CapsLock').classList.add('pressed');
+  const rows = keyboard.children;
+  for (let row = 0; row < rows.length; row += 1) {
+    const keys = rows[row].children;
+    for (let key = 0; key < keys.length; key += 1) {
+      if (keys[key].querySelectorAll('.lower.active').length !== 0) {
+        keys[key].querySelectorAll('.upper').forEach((item) => item.classList.add('active'));
+        keys[key].querySelectorAll('.lower').forEach((item) => item.classList.remove('active'));
+      } else {
+        keys[key].querySelectorAll('.upper').forEach((item) => item.classList.remove('active'));
+        keys[key].querySelectorAll('.lower').forEach((item) => item.classList.add('active'));
+      }
+    }
+  }
+}
+
+function doUncaps() {
+  document.querySelector('.CapsLock').classList.remove('pressed');
+  isCaps = false;
+  const rows = keyboard.children;
+  for (let row = 0; row < rows.length; row += 1) {
+    const keys = rows[row].children;
+    for (let key = 0; key < keys.length; key += 1) {
+      if (keys[key].querySelectorAll('.lower.active').length !== 0) {
+        keys[key].querySelectorAll('.upper').forEach((item) => item.classList.add('active'));
+        keys[key].querySelectorAll('.lower').forEach((item) => item.classList.remove('active'));
+      } else {
+        keys[key].querySelectorAll('.upper').forEach((item) => item.classList.remove('active'));
+        keys[key].querySelectorAll('.lower').forEach((item) => item.classList.add('active'));
+      }
+    }
+  }
+}
+
+function changeLang() {
+  const rows = keyboard.children;
+  for (let row = 0; row < rows.length; row += 1) {
+    const keys = rows[row].children;
+    for (let key = 0; key < keys.length; key += 1) {
+      if (keys[key].querySelectorAll('.ru.on').length) {
+        keys[key].querySelectorAll('.en').forEach((item) => item.classList.add('on'));
+        keys[key].querySelectorAll('.ru').forEach((item) => item.classList.remove('on'));
+        window.localStorage.setItem('lang', 'en');
+      } else {
+        keys[key].querySelectorAll('.en').forEach((item) => item.classList.remove('on'));
+        keys[key].querySelectorAll('.ru').forEach((item) => item.classList.add('on'));
+        window.localStorage.setItem('lang', 'ru');
+      }
+    }
+  }
+}
+
+if (window.localStorage.getItem('lang') === 'en') {
+  changeLang();
+}
+
+// adding mousedown listeners for clicks on virtual keyboard
+document.addEventListener('mousedown', (e) => {
+  if (e.target.classList.contains('Tab')
```

Comment:
Try to move  e.target.classList.contains into a variable, for example:
const isTargetClass = (className) => e.target.classList.contains(className);
the same with the parentElement